torch/distributed/elastic/utils/data/elastic_distributed_sampler.py

```diff
@@ -34,11 +39,17 @@ class ElasticDistributedSampler(DistributedSampler):
         start_index (optional):  Which index of the dataset to start sampling from
     """
 
-    def __init__(self, dataset, num_replicas=None, rank=None, start_index=0):
+    def __init__(
+        self,
+        dataset: Dataset[T],
+        num_replicas: Optional[int] = None,
+        rank: Optional[int] = None,
+        start_index: int = 0,
+    ):
         super().__init__(dataset=dataset, num_replicas=num_replicas, rank=rank)
-        if start_index >= len(dataset):
+        if start_index >= len(dataset):  # type: ignore[arg-type]
```

Comment:
@bobrenjc93 You need to add an assertion here if the dataset isn't Sized? Is there anway we can check an intersection of Dataset and Sized here?

Reply:
I cancelled the merge and will fix before landing. thx!